client/control/components/Header.js

```diff
@@ -32,4 +52,56 @@ Header.propTypes = {
   currentLocation: pt.string.isRequired,
   routes: pt.array.isRequired,
   params: pt.object.isRequired,
+  token: pt.shape({ key: pt.string.isRequired }),
+  user: pt.object.isRequired,
 };
+
+export default connect(
+  (state, ownProps) => ({
+    ...ownProps,
+    token: state.auth.token,
+  }),
+)(Header);
+
+
+class APITokenExpander extends Component {
+  static propTypes = {
+    token: pt.shape({ key: pt.string }),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: false };
+    this.toggleExpand = ::this.toggleExpand;
+  }
+
+  toggleExpand() {
+    this.setState(state => ({ ...state, expanded: !state.expanded }));
```

Comment:
The `...state` thing was a mistake. I had remembered that in order for state changes to be atomic they had to use the function form if they were based on the previous state, but I can't find any reference to that anymore, or reproduce it in a test, so I guess I'm just misremembering.